crates/swc_ecma_transforms_compat/tests/es2015_regenerator.rs

```diff
@@ -1169,6 +1169,410 @@ test_exec!(
+                                        b
+                                    ) {
+                                        return regeneratorRuntime.wrap(
+                                            function _callee$(_ctx) {
+                                                while (1)
+                                                    switch (_ctx.prev = _ctx.next) {
+                                                        case 0:
+                                                            return _ctx.abrupt(
+                                                                "return",
+                                                                a + b
+                                                            );
+                                                        case 1:
+                                                        case "end":
+                                                            return _ctx.stop();
+                                                    }
+                                            },
+                                            _callee
+                                        );
+                                    })
+                                );
+                                return _bar1.apply(this, arguments);
+                            };
+                            return _ctx1.abrupt("return", bar1);
+                        case 7:
+                        case "end":
+                            return _ctx1.stop();
+                    }
+            }, _callee1);
+        })
+    );
+    return _foo.apply(this, arguments);
+}
+foo(1)
+    .then((t) => t(2))
+    .then(console.log);
+"#
+);
+
+test_exec!(
+    Syntax::default(),
+    |_| chain!(async_to_generator(), tr(())),
+    hoist_function_in_async_issue_2556_7,
+    r#"
+async function foo(a) {
+    return bar1;
+
+    const bar1 = "bar1";
+    console.log("dead code should not output", bar1);
+
+    async function bar1(b) {
+        return a + b;
+    }
+}
+
+return foo(1)
+    .then((t) => t(2))
+    .then((result) => {
+        expect(result).toBe(3);
+    });
+"#
+);
+
+test!(
+    Syntax::default(),
+    |_| chain!(async_to_generator(), tr(())),
+    hoist_function_in_async_issue_2556_8,
+    r#"
+var fib = function fib() {
+    return 42;
+};
+async function init() {
+    return fib;
+
+    const fib = 55;
+    console.log("dead code", fib);
+
+    async function fib(n) {
+        if (n <= 1) {
+            return n;
+        }
+        const x = await fib(n - 1);
+        const y = await fib(n - 2);
+        return x + y;
+    }
+}
+"#,
+    r#"
+var regeneratorRuntime = require("regenerator-runtime");
+var fib = function fib() {
+    return 42;
+};
+function init() {
+    return _init.apply(this, arguments);
+}
+function _init() {
+    _init = _asyncToGenerator(
+        regeneratorRuntime.mark(function _callee1() {
+            var fib1, fib1, _fib;
```

Comment:
Hi, @kdy1 ,
The Transformed code is here.
You're right, the variable names are a bit buggy.